charlib/characterizer/cell.py

```
@@ -74,20 +74,31 @@ def __init__(self, name: str, supply_nodes: dict,  **cell_config):
-        functions = dict()
-        inputs = set()
-        outputs = set()
         def _parse_expression(expression):
             # Helper function for parsing expressions of the form Y=A&B etc.
             lhs, rhs = expression.split('=')
             output = ''.join([c for c in lhs if c.isalnum() or c == '_'])
             if not parse_logic(rhs):
                 raise ValueError(f'Could not parse expression "{expression}"')
             return output, rhs.strip()
+
+        functions = dict()
```

Ruff (0.15.21)

[warning] Unnecessary `dict()` call (rewrite as a literal)

Rewrite as a literal

(C408)